src/purchases.ts

```diff
@@ -0,0 +1,716 @@
+    }
+
+    /**
+     * Subscriber attribute associated with the Facebook SDK Anonymous Id for the user
+     * Recommended for the RevenueCat Facebook integration
+     *
+     * @param fbAnonymousID Empty String or null will delete the subscriber attribute.
+     */
+    public static setFBAnonymousID(fbAnonymousID: string | null): void {
+        RNPurchases.setFBAnonymousID(fbAnonymousID);
+    }
+
+    /**
+     * Subscriber attribute associated with the mParticle Id for the user
+     * Recommended for the RevenueCat mParticle integration
+     *
+     * @param mparticleID Empty String or null will delete the subscriber attribute.
+     */
+    public static setMparticleID(mparticleID: string | null): void {
+        RNPurchases.setMparticleID(mparticleID);
+    }
+
+    /**
+     * Subscriber attribute associated with the OneSignal Player Id for the user
+     * Required for the RevenueCat OneSignal integration
+     *
+     * @param onesignalID Empty String or null will delete the subscriber attribute.
+     */
+    public static setOnesignalID(onesignalID: string | null): void {
+        RNPurchases.setOnesignalID(onesignalID);
+    }
+
+    /**
+     * Subscriber attribute associated with the install media source for the user
+     *
+     * @param mediaSource Empty String or null will delete the subscriber attribute.
+     */
+    public static setMediaSource(mediaSource: string | null): void {
+        RNPurchases.setMediaSource(mediaSource);
+    }
+
+    /**
+     * Subscriber attribute associated with the install campaign for the user
+     *
+     * @param campaign Empty String or null will delete the subscriber attribute.
+     */
+    public static setCampaign(campaign: string | null): void {
+        RNPurchases.setCampaign(campaign);
+    }
+
+    /**
+     * Subscriber attribute associated with the install ad group for the user
+     *
+     * @param adGroup Empty String or null will delete the subscriber attribute.
+     */
+    public static setAdGroup(adGroup: string | null): void {
+        RNPurchases.setAdGroup(adGroup);
+    }
+
+    /**
+     * Subscriber attribute associated with the install ad for the user
+     *
+     * @param ad Empty String or null will delete the subscriber attribute.
+     */
+    public static setAd(ad: string | null): void {
+        RNPurchases.setAd(ad);
+    }
+
+    /**
+     * Subscriber attribute associated with the install keyword for the user
+     *
+     * @param keyword Empty String or null will delete the subscriber attribute.
+     */
+    public static setKeyword(keyword: string | null): void {
+        RNPurchases.setKeyword(keyword);
+    }
+
+    /**
+     * Subscriber attribute associated with the install ad creative for the user
+     *
+     * @param creative Empty String or null will delete the subscriber attribute.
+     */
+    public static setCreative(creative: string | null): void {
+        RNPurchases.setCreative(creative);
+    }
+
+}
```

Comment:
this is still a pretty big file, and I think we could probably extract all of the subscriber attributes stuff into its own module, but figured this was a big enough refactor already